### DISCUSSION TITLE
Export definitions in yaml format (for Kubernetes)

### DISCUSSION THREAD

**Discussion by pvladMQ** [rabbitmq/rabbitmq-server#12977] [Ideas]
### Is your feature request related to a problem? Please describe.

When migrating from Linux/Windows RabbitMQ installation to a Kubernetes environment, there is no way to export definitions so they can be used with the Messaging topology operator. 

### Describe the solution you'd like

A way to export definitions in Yaml format so artifacts can be created with the Messaging topology operator; probably a good idea would be an update on `export_definitions` or another similar function. 

### Describe alternatives you've considered

I have a python script that does that, but not all organizations accept 3rd parties scripts to be used.

**Comment by pvladMQ** [comment]
I want to work on this on my own, opened the issue so I can keep track of my progress and give the chance to debate over the feature.

**Comment by michaelklishin** [comment]
@pvladMQ I think this should be an external tool instead. You can use any RabbitMQ HTTP API client to fetch the definitions, transform them (since there likely will be differences with the YAML structure used by the Message Topology Operator) and serialize to YAML.

I don't see why it should be a RabbitMQ feature. Most users won't need to export definitions as YAML since RabbitMQ does not accept importing them from YAML and expanding an already complex part to also support YAML is unlikely to attract much interest from the core team.
